File: Helpers/src/main/java/com/dg/helpers/LongHelper.java
package com.dg.helpers;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */
public class LongHelper
{
    public static Long withObject(Object value)
    {
        if (value == null) return null;
        else if (value instanceof Long) return (Long)value;
        else if (value instanceof Integer) return (long) (Integer) value;
        else if (value instanceof Short) return (long) (Short) value;
        else if (value instanceof String)
        {
            try { return Long.parseLong((String) value); }
            catch (Exception ignored) { }
        }
        return null;
    }

    public static Long withObject(Object value, long defaultValue)
    {
        if (value == null) return defaultValue;
        else if (value instanceof Long) return (Long)value;
        else if (value instanceof Integer) return (long) (Integer) value;
        else if (value instanceof Short) return (long) (Short) value;
        else if (value instanceof String)
        {
            try { return Long.parseLong((String) value); }
            catch (Exception ignored) { }
        }
        return defaultValue;
    }

    public static long[] toArray(Long[] inArray)
    {
        long[] array = new long[inArray.length];
        for (int i = 0, len = inArray.length; i < len; i++)
        {
            Long value = inArray[i];

            if (value != null)
            {
                array[i] = value;
            }
        }
        return array;
    }
}

File: Helpers/src/main/java/com/dg/helpers/LongHelper.kt
package com.dg.helpers

@Suppress("unused")
object LongHelper
{
    @Suppress("MemberVisibilityCanBePrivate")
    fun withObject(value: Any?): Long?
    {
        when (value)
        {
            null -> return null
            is Long -> return value
            is Int -> return value.toLong()
            is Short -> return value.toLong()
            is String -> try
            {
                return java.lang.Long.parseLong(value)
            }
            catch (ignored: Exception)
            {
            }
        }
        return null
    }

    fun withObject(value: Any?, defaultValue: Long): Long
    {
        return withObject(value) ?: defaultValue
    }

    fun toArray(inArray: Array<Long?>): LongArray
    {
        val array = LongArray(inArray.size)
        var i = 0
        val len = inArray.size
        while (i < len)
        {
            val value = inArray[i]

            if (value != null)
            {
                array[i] = value
            }
            i++
        }
        return array
    }
}
